src/main/groovy/com/bmuschko/gradle/docker/utils/MainClassFinder.groovy

```diff
@@ -123,7 +123,7 @@ class MainClassFinder {
         stack.push(rootFolder)
         while (!stack.isEmpty()) {
             File file = stack.pop()
-            if (file.isFile()) {
+            if (file.isFile() && file.name.endsWith('.class')) {
```

Comment:
There's already a constant for `.class` named `DOT_CLASS`. I'll clean it up after the merge.

Reply:
Something went wrong there, this shouldn't be in there at all. The fix in the FileFilter accept methods is sufficient.

Reply:
I see that in commit 08fd39bfe22e01f16f66f8a5640c0ea180fa8929 you reverted pathname back to file. It should be pathname in both accept methods. And line 126 should just be `if (file.isFile()) {`

The current code in master also works correctly, but is not the cleanest solution.

I've updated my pull request a couple of times and I think I accidentally reintroduced the initial solution.

Reply:
Can you elaborate on "clean solution" here? I didn't have another deep look but can you compare both approaches with pros/cons?

Reply:
I've added a comment to the code, I hope I'm being clear.